packages/javascript/bh-shared-ui/src/components/EnvironmentSelectPanel/EnvironmentSelectPanel.tsx

```diff
@@ -0,0 +1,222 @@
+                        <Input
+                            variant='underlined'
+                            className='w-full ml-3 border-b-0 focus:!border-b-0 hover:!border-b-0 dark:focus:!border-b-0 dark:hover:!border-b-0'
+                            id='search'
+                            type='text'
+                            placeholder='Search'
+                            onChange={(e) => {
+                                setSearchInput(e.target.value);
+                            }}
+                        />
+                    </div>
+                    <div
+                        className='flex flex-row ml-4 mt-6 mb-2 items-center'
+                        data-testid='create-user-dialog_select-all-environments-checkbox-div'>
+                        <FormField
+                            name='all_environments'
+                            control={form.control}
+                            render={() => (
+                                <FormItem className='flex flex-row items-center'>
+                                    <Checkbox
+                                        checked={areAllEnvironmentsSelected || areAllEnvironmentsIndeterminate}
+                                        id='allEnvironments'
+                                        onCheckedChange={handleSelectAllEnvironmentsChange}
+                                        className={cn(
+                                            areAllEnvironmentsSelected &&
+                                                '!bg-primary border-neutral-dark-1 dark:!bg-neutral-light-2'
+                                        )}
+                                        icon={
+                                            areAllEnvironmentsIndeterminate && (
+                                                <Minus
+                                                    className='h-full w-full bg-neutral-light-2 text-neutral-dark-1 dark:bg-neutral-dark-2 dark:text-neutral-light-2'
+                                                    absoluteStrokeWidth={true}
+                                                    strokeWidth={3}
+                                                />
+                                            )
+                                        }
+                                        data-testid='create-user-dialog_select-all-environments-checkbox'
+                                    />
+                                    <FormLabel
+                                        htmlFor='allEnvironments'
+                                        className='ml-3 w-full cursor-pointer font-normal'>
+                                        Select All Environments
+                                    </FormLabel>
+                                </FormItem>
+                            )}
+                        />
+                    </div>
+                    <div className='flex flex-col' data-testid='create-user-dialog_environments-checkboxes-div'>
```

Comment:
I tested this with 50 domains and the list of environments expands past the form, i attached a wee video example. Perhaps we lock the height of of the `EnvironmentSelectPanel` to the height of the form?

https://github.com/user-attachments/assets/18a537fe-176e-41f2-baca-fc919347002a